### DISCUSSION TITLE
Comparable tool to truffle dashboard?

### DISCUSSION THREAD

**Discussion by critesjosh** [ConsenSys-archive/truffle#6213] [General Q&A] [answered]
I really like the truffle dashboard tool for building CLI apps that use metamask. Are there any comparable tools?

**Comment by gnidan** [comment, marked as answer]
It doesn't work with MetaMask, per se, but you can use Frame.sh, which exposes its own port that functions about the same as Truffle Dashboard. Not sure if this feature is in their docs yet, but they talk about it in [this tweet thread](https://twitter.com/0xframe/status/1680992123968163841)

**Reply by critesjosh** [reply]
I wasn't aware of this, this is great!

How difficult was it to maintain the dashboard functionality? I imagine it wouldn't be too difficult to create a standalone package that has this functionality--its just a web server that is relaying information.

**Reply by gnidan** [reply]
Nah, not too hard. It's really just a glorified proxy/message bus, but there's a bit of trickiness when you want to make a robust system that doesn't drop messages. The requirements get interesting when you consider what to do about multiple browser tabs or multiple request senders... ideally you want requests to go to all open tabs (since people might forget to close tabs and lose them), and of course responses need to go back to their correct origin.

**Reply by critesjosh** [reply]
Ah thanks. I'm looking to build a dev tool, so making it super robust may not be worth it.
